### PR TITLE
Return empty list of provided type

### PR DESCRIPTION
**Pull request by adamschaub**
Addresses "expected type" error in #222 

For any pointer-type values (maps, slices), a base value of `nil` should be converted to an an initialized (but empty, possibly nil) equivalent (via reflection). 

Make sure it works for:
- [x] Maps/objects
- [x] Slices of all types

### CODE REVIEW

File: model/convert.go [hunk 1]
```diff
@@ -14,10 +13,12 @@ func ConvertIO(a interface{}, t string, plug Plugin) (interface{}, bool) {
 		typ := reflect.ValueOf(a)
 		var l []interface{}
 		if typ.Kind() != reflect.Slice {
-			if v, ok := ConvertIO(a, out, plug); ok {
-				l = append(l, v)
-			} else {
-				return nil, false
+			if a != nil {
```
Comment (helik) [inline model/convert.go]: So what happens if a == nil? It returns a, false in line 42?

File: model/convert_test.go [hunk 2]
```diff
@@ -355,12 +374,11 @@ func TestConvertPluginIO(t *testing.T) {
 			can: true,
 		},
 		// Ones that don't work
```
Comment (helik) [inline model/convert_test.go]: Should this comment be moved since the next test does work?

File: model/convert.go (hunk 1 above)
Comment (adamschaub) [inline model/convert.go]: it returns `final, true` if it correctly passes `toListOfType`

Comment (helik) [inline model/convert.go]: ah gotcha, didn't see that

**Pull request merged**